### PR TITLE
docs: add v8 release announcement blog post

### LINKED ISSUE #9596: Docs: Add final announcement blog post for v8
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

v8 is pretty close to launch! We'll want to:

* Update https://typescript-eslint.io/blog/announcing-typescript-eslint-v8-beta to reflect the latest changes:
  * Any new breaking changes & important bits that landed
  * Refreshing the list of community contributors
* Add a new https://typescript-eslint.io/blog/announcing-typescript-eslint-v8

Depending on how #9571, we _may_ need to add some coaching language around performance. For example, it might be that superior perf with project service is blocked on followup PRs on our end and/or TypeScript's.

### Affected URL(s)

https://typescript-eslint.io/blog/announcing-typescript-eslint-v8

💖

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## PR Checklist

- [x] Addresses an existing open issue: fixes #9596
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

Does a similar approach to v6's announcement:

1. Adds a copy & past of the beta post to a new stable one, with names & paths updated
2. Updates the beta post to point to the stable one
3. Updates the code snippet showing the config diff

Also adds in the list of community projects & plugins that tried out v8 with their maintainers under thanks.

<details>
<summary>Code that generated v7->v8 config diffs</summary>

```json title="package.json"
{
  "dependencies": {
    "@typescript-eslint/eslint-plugin-7": "npm:@typescript-eslint/eslint-plugin@7",
    "@typescript-eslint/eslint-plugin-8": "npm:@typescript-eslint/eslint-plugin@rc-v8"
  },
  "type": "module"
}
```

```js
import tseslint7 from '@typescript-eslint/eslint-plugin-7';
import tseslint8 from '@typescript-eslint/eslint-plugin-8';

function createDiffPatch(v7, v8) {
  const v7Keys = new Set(Object.keys(v7));
  const v8Keys = new Set(Object.keys(v8));
  const output = ['{'];

  for (const key of Array.from(new Set([...v7Keys, ...v8Keys])).sort((a, b) =>
    trimSlash(a).localeCompare(trimSlash(b)),
  )) {
    const prefix = v7Keys.has(key) ? (v8Keys.has(key) ? ' ' : '-') : '+';

    output.push(`${prefix}  '${key}': '...',`);
  }

  output.push('}');

  return output.join('\n');
}

function trimSlash(text) {
  return text.startsWith('@typescript-eslint/')
    ? text.slice('@typescript-eslint/'.length)
    : text;
}

const configNames = [
  'recommended',
  'recommended-type-checked',
  'strict',
  'strict-type-checked',
  'stylistic',
  'stylistic-type-checked',
];

for (const configName of configNames) {
  console.log(configName);
  console.log(
    createDiffPatch(
      tseslint7.configs[configName].rules,
      tseslint8.configs[configName].rules,
    ),
  );
}
```

</details>

💖

### CODE REVIEW

File: packages/website/blog/2024-07-31-announcing-typescript-eslint-v8.md [hunk 1]
```diff
@@ -0,0 +1,816 @@
...
+  console.log(
+    createDiffPatch(
+      tseslint7.configs[configName].rules,
+      tseslint8.configs[configName].rules,
+    ),
+  );
+}
+```
+
+</details>
```
Comment (bradzacher) [inline packages/website/blog/2024-07-31-announcing-typescript-eslint-v8.md:509]: do we want to include this in the blog post? seems like a distraction.

File: packages/website/blog/2024-07-31-announcing-typescript-eslint-v8.md [hunk 2]
```diff
@@ -0,0 +1,816 @@
...
+
+- [ESLint > Migrate to 9.0.0 > `FlatRuleTester` is now `RuleTester`](https://eslint.org/docs/latest/use/migrate-to-9.0.0#-flatruletester-is-now-ruletester)
+- [typescript-eslint > feat(rule-tester): switched to flat config](https://github.com/typescript-eslint/typescript-eslint/pull/9603)
+
+### Support for multi-pass fixes in `RuleTester`
+
+One limitation of ESLint's `RuleTester` is that it is not possible to verify the individual applied fixes when a rule provides multiple rounds of fixes.
+[ESLint's `RuleTester` applies only the first fix](https://eslint.org/docs/latest/integrate/nodejs-api#testing-fixes) when there is conflict between two fixes.
+
+In typescript-eslint v8, the typescript-eslint `RuleTester` tries to apply all possible fixes for each test case.
```
Comment (bradzacher) [inline packages/website/blog/2024-07-31-announcing-typescript-eslint-v8.md:729]:
```suggestion
In typescript-eslint v8, our `RuleTester` tries to apply all possible fixes for each test case.
```

nit cos it looks weird seeing the name twice in a row

**Review by bradzacher — DISMISSED** [review]

File: packages/website/blog/2024-07-31-announcing-typescript-eslint-v8.md (hunk 1 above)
Comment (JoshuaKGoldberg) [reply packages/website/blog/2024-07-31-announcing-typescript-eslint-v8.md:509]: Heh, yeah, good point. It was more for me as the author than anybody else.

**Comment by JoshuaKGoldberg** [comment]
Noting that we'll want to merge this on Wednesday & trigger a manual website release.

**Pull request merged**